homeassistant/components/asuswrt/config_flow.py

```diff
@@ -83,14 +83,17 @@ def _show_setup_form(
         schema = {
             vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
             vol.Required(CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")): str,
-            vol.Optional(CONF_PASSWORD)
-            if self.show_advanced_options
-            else vol.Required(CONF_PASSWORD): str,
-            vol.Required(CONF_PROTOCOL, default=PROTOCOL_SSH): vol.In(
-                {PROTOCOL_SSH: "SSH", PROTOCOL_TELNET: "Telnet"}
-            ),
         }
 
+        if self.show_advanced_options:
+            schema[vol.Optional(CONF_PASSWORD)] = str
+        else:
+            schema[vol.Required(CONF_PASSWORD)] = str
```

Comment:
We can set password as required as default when first defining the schema and if we're in advanced mode we overwrite it as optional.

Reply:
Sorry but I don't not understand how to do this. This is `dict`, how I should replace the key `vol.Required` with the new `vol.Optional` maintaining the position in the `dict`?

Reply:
Just set it as we already do above. We don't need change that, I believe.

Reply:
No, doesn't work, I tested this. If both `Required` and `Optional` are set for the same variable. it use the first (it remain `Required`).

Reply:
Then we can use `Schema.extend`.

Reply:
Ok, I try.

Reply:
Also `Schema.extend` do not solve, because update the field but move it to lower position. What about the solution in my last commit?